arch/x86/core/userspace.c

```diff
@@ -96,6 +96,13 @@ FUNC_NORETURN void arch_user_mode_enter(k_thread_entry_t user_entry,
 				      _current->stack_info.size -
 				      _current->stack_info.delta);
 
+	/* x86_64 SysV ABI requires 16 byte stack alignment, which
```

Comment:
Wondering if this should be enabled only for `CONFIG_X86_64 || CONFIG_SSE` since 32-bit does not seem to require this.

Reply:
That's an excellent point, will fix.  But not SSE I don't think: in 32 bit mode there's no stack alignment requirement so if the compiler did want to autovectorize it would be responsible for aligning the stack on its own and wouldn't need this anyway.